config/tools/config_generator.hpp

```diff
@@ -0,0 +1,179 @@
+        static const inline auto m_secp
+            = std::unique_ptr<secp256k1_context,
+                              decltype(&secp256k1_context_destroy)>(
+                secp256k1_context_create(SECP256K1_CONTEXT_SIGN),
+                &secp256k1_context_destroy);
+
+        // Where the newly created configuration parameters will be stored as
+        // we go along generating them
+        std::stringstream m_new_config;
+
+        std::map<std::string, value_t> template_options;
+
+        // Calculate shard coverage
+        void calculate_shard_coverage(size_t num_shards, size_t shard_size);
+        // Helper for calculating shard id coverage
+        void shard_bookkeeping(const std::vector<size_t>& array_total,
+                               size_t shard_id);
+        // Get random value based on mean and standard deviation
+        [[nodiscard]] auto calculate_normal_distribution_point(size_t mean,
+                                                               double std_dev)
+            -> double;
+        // This is not a failsafe because we are simply generating a
+        // configuration file here, however, it is still good to check that the
+        // port is available
+        [[nodiscard]] auto get_open_port() -> unsigned short;
+        // Create Private/Public key pair repeatably (repeatable from run to
+        // run)
+        [[nodiscard]] auto create_repeatable_key_pair()
+            -> std::pair<std::string, std::string>;
+        // Create Private/Public key pair randomly (NOT repeatable from run to
+        // run)
+        [[nodiscard]] auto create_random_key_pair()
+            -> std::pair<std::string, std::string>;
+        // Create Private/Public key pair
+        [[nodiscard]] auto create_key_pair()
+            -> std::pair<std::string, std::string>;
+        // Parse value as int, double or string from config file
+        [[nodiscard]] auto parse_value(const std::string& value,
+                                       bool keep_quotes) -> value_t;
+        [[nodiscard]] auto get_param_from_template_file(
+            std::string option,
+            std::map<std::string, std::string>& config_map)
+            -> std::variant<std::string, size_t, double>;
+        void set_param_to_config_file(std::string key, std::string value);
+        void set_param_to_config_file(std::string key, double value);
+        void set_param_to_config_file(std::string key, size_t value);
+        // Helper to set proper log level for either Two-Phase Commit or
+        // Atomizer components
+        void set_log_level(std::string key, std::string& log_level);
+        // Method to create all the related components for generated config
+        // file
+        [[maybe_unused]] auto create_component(const char* type,
+                                               value_t component_count,
+                                               bool create_2pc) -> std::string;
+
+        // Create 2PC shards
```

Comment:
Much of these comments are probably not necessary (they mirror the method name exactly), but when adding documentation-comments, adding an additional slash (`///`) will make them visible to doxygen.

Reply:
Removed them.

DONE